### DISCUSSION TITLE
Minimal Dockerfile for embedding FrankenPHP in a Go app?

### DISCUSSION THREAD

**Discussion by jasonmccallister** [php/frankenphp#531] [Q&A]
I'm not too familiar with CGO, but is there a recommended `Dockerfile` to use for embedding FrankenPHP into a Go application? I'm using the example code from the Go docs but pasting here for clarity:

```
package main

import (
	"log"
	"net/http"

	"github.com/dunglas/frankenphp"
)

func main() {
	if err := frankenphp.Init(); err != nil {
		panic(err)
	}
	defer frankenphp.Shutdown()

	http.HandleFunc("/", func(w http.ResponseWriter, r *http.Request) {
		req, err := frankenphp.NewRequestWithContext(r, frankenphp.WithRequestDocumentRoot("/path/to/document/root", false))
		if err != nil {
			panic(err)
		}

		if err := frankenphp.ServeHTTP(w, req); err != nil {
			panic(err)
		}
	})
	log.Fatal(http.ListenAndServe(":8080", nil))
}
```

> If I can get this running before getting a response, I will try to update this with an example and minimal `Dockerfile`.

**Comment by aszenz** [comment]
I'm also interested in embedding use case, did u figure out how to build it

**Comment by dunglas** [comment]
You can adapt our dev Dockerfile: https://github.com/dunglas/frankenphp/blob/main/dev.Dockerfile

**Reply by aszenz** [reply]
Thanks, it works, is their an easy way  to add extensions to the php build process

**Reply by aszenz** [reply]
@dunglas Friendly ping, i would like to explore the embedding use case but I need to create a php lib with extensions, the docs don't cover the embedding php as lib usecase

**Reply by withinboredom** [reply]
@aszenz I have an example CLI here (it's kind of a mess/proof-of-concept): https://github.com/bottledcode/durable-php/tree/v2/cli that embeds frankenphp into a CLI application. To install php extensions, you install the extensions like normal—or include them in the static builder.
